sdk/terraform/Azure.ResourceManager.Terraform/api/Azure.ResourceManager.Terraform.netstandard2.0.cs

```diff
@@ -0,0 +1,146 @@
+        Azure.ResourceManager.Terraform.Models.BaseExportModel System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.Terraform.Models.BaseExportModel>.Create(System.BinaryData data, System.ClientModel.Primitives.ModelReaderWriterOptions options) { throw null; }
+        string System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.Terraform.Models.BaseExportModel>.GetFormatFromOptions(System.ClientModel.Primitives.ModelReaderWriterOptions options) { throw null; }
+        System.BinaryData System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.Terraform.Models.BaseExportModel>.Write(System.ClientModel.Primitives.ModelReaderWriterOptions options) { throw null; }
+    }
+    public partial class ExportQuery : Azure.ResourceManager.Terraform.Models.BaseExportModel, System.ClientModel.Primitives.IJsonModel<Azure.ResourceManager.Terraform.Models.ExportQuery>, System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.Terraform.Models.ExportQuery>
+    {
+        public ExportQuery(string query) { }
+        public string NamePattern { get { throw null; } set { } }
+        public string Query { get { throw null; } }
+        public bool? Recursive { get { throw null; } set { } }
+        protected override void JsonModelWriteCore(System.Text.Json.Utf8JsonWriter writer, System.ClientModel.Primitives.ModelReaderWriterOptions options) { }
+        Azure.ResourceManager.Terraform.Models.ExportQuery System.ClientModel.Primitives.IJsonModel<Azure.ResourceManager.Terraform.Models.ExportQuery>.Create(ref System.Text.Json.Utf8JsonReader reader, System.ClientModel.Primitives.ModelReaderWriterOptions options) { throw null; }
+        void System.ClientModel.Primitives.IJsonModel<Azure.ResourceManager.Terraform.Models.ExportQuery>.Write(System.Text.Json.Utf8JsonWriter writer, System.ClientModel.Primitives.ModelReaderWriterOptions options) { }
+        Azure.ResourceManager.Terraform.Models.ExportQuery System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.Terraform.Models.ExportQuery>.Create(System.BinaryData data, System.ClientModel.Primitives.ModelReaderWriterOptions options) { throw null; }
+        string System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.Terraform.Models.ExportQuery>.GetFormatFromOptions(System.ClientModel.Primitives.ModelReaderWriterOptions options) { throw null; }
+        System.BinaryData System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.Terraform.Models.ExportQuery>.Write(System.ClientModel.Primitives.ModelReaderWriterOptions options) { throw null; }
+    }
+    public partial class ExportResource : Azure.ResourceManager.Terraform.Models.BaseExportModel, System.ClientModel.Primitives.IJsonModel<Azure.ResourceManager.Terraform.Models.ExportResource>, System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.Terraform.Models.ExportResource>
```

Comment:
this name reads like a verb phrase.
I think this means the information for the service to export these resources into terraform hcl configs.
Therefore how about `ExportFromResourceContent` or `ExportContentFromResource` or `ExportContentFromResources`

Reply:
I prefer these terms to stay as is because they map tightly to the argument names in CLI. Check:

```
$  az  azure-terraform  export-terraform --help
...
ExportParameter Arguments
    --export-query          : Export parameter for resources queried by ARG (Azure Resource Graph)
                              Support shorthand-syntax, json-file and yaml-file. Try "??" to show
                              more.
    --export-resource       : Export parameter for single resources  Support shorthand-syntax, json-
                              file and yaml-file. Try "??" to show more.
    --export-resource-group : Export parameter for a resource group  Support shorthand-syntax, json-
                              file and yaml-file. Try "??" to show more.
```

cc: @magodo

Reply:
type that ends with `Resource` has a special meaning in .Net SDK - it should be a `ArmResource`, but there it is not.
Therefore this is not a very good name here.
Also I noticed you added this name into the `exception list`, which we should not do unless it appears in an old RP before we have that test case.

Reply:
I have renamed ExportResource to ExportResourceTerraform via client.tsp. See: ⁠[d16ae2617a](https://github.com/Azure/azure-rest-api-specs/pull/31198/commits/d16ae2617a752878c5030e45467f296e81974d3c). InheritanceCheckTest passes after the exclusion is removed (corresponding commit in dotnet sdk: ⁠[3963ac662bd](https://github.com/Azure/azure-sdk-for-net/pull/46301/commits/3963ac662bd063bbee5541d35124fcb3eb61d20c)).